NeoMathEngine/src/GPU/CUDA/Kernels/CudaDnnDropoutKernels.h

```diff
@@ -33,4 +34,43 @@ __global__ void ChannelLastBlobSpatialDropoutKernel( const float* __restrict__ i
 	}
 }
 
+__global__ void RandomMatrixDropout( const float* __restrict__ first, int firstHeight,
+	int firstWidth, float* res, int seed, float forwardRate )
+{
+	const unsigned int threshold = forwardRate * UINT_MAX;
+	int row;
+	int col;
+	if (GetCudaTaskIndex2D( firstHeight, ( firstWidth + 3 ) / 4, row, col ) ) {
+		CCudaRandom random(seed);
+		random.Skip(col);
+		col *= 4;
+		const int index = row * firstWidth + col;
+
+		CIntArray<4> generated = random.Next();
+		for (int j = 0; j < 4 && col + j < firstWidth; ++j) {
+			res[index + j] = (generated[j] <= threshold) ? (first[index + j] / forwardRate ) : 0.f;
+		}
+	}
+}
+
+__global__ void RandomSpatialDropout( const float* __restrict__ input, float* res, int inputObjectCount,
+	int inputObjectSize, int maskObjectCount, int maskObjectSize, int seed, float forwardRate )
+{
+	const unsigned int threshold = forwardRate * UINT_MAX;
+	int obj;
+	int row;
+	int col;
+	if ( GetCudaTaskIndex3D(inputObjectCount, inputObjectSize / maskObjectSize, maskObjectSize, obj, row, col ) ) {
```

Comment:
_minor_ _codestyle_ There should be no space between `if` and `(` (same rule applies to anything `for(` `while(` etc.